docs/sources/alerting/_index.md

```diff
@@ -0,0 +1,260 @@
+
+One option to scale the Ruler is by scaling it horizontally. However, with multiple Ruler instances running they will need to coordinate to determine which instance will evaluate which rule. Similar to the ingesters, the Rulers establish a hash ring to divide up the responsibilities of evaluating rules.
+
+The possible configurations are listed fully in the configuration [docs](https://grafana.com/docs/loki/latest/configuration/), but in order to shard rules across multiple Rulers, the rules API must be enabled via flag (`-experimental.Ruler.enable-api`) or config file parameter. Secondly, the Ruler requires it's own ring be configured. From there the Rulers will shard and handle the division of rules automatically. Unlike ingesters, Rulers do not hand over responsibility: all rules are re-sharded randomly every time a Ruler is added to or removed from the ring.
+
+A full Ruler config example is:
+
+```yaml
+Ruler:
+    alertmanager_url: <alertmanager_endpoint>
+    enable_alertmanager_v2: true
+    enable_api: true
+    enable_sharding: true
+    ring:
+        kvstore:
+            consul:
+                host: consul.loki-dev.svc.cluster.local:8500
+            store: consul
+    rule_path: /tmp/rules
+    storage:
+        gcs:
+            bucket_name: <loki-rules-bucket>
+```
+
+## Ruler storage
+
+The Ruler supports six kinds of storage: configdb, azure, gcs, s3, swift, and local. Most kinds of storage work with the sharded Ruler configuration in an obvious way, i.e. configure all Rulers to use the same backend.
+
+The local implementation reads the rule files off of the local filesystem. This is a read only backend that does not support the creation and deletion of rules through [the API](https://grafana.com/docs/loki/latest/api/#Ruler). Despite the fact that it reads the local filesystem this method can still be used in a sharded Ruler configuration if the operator takes care to load the same rules to every Ruler. For instance this could be accomplished by mounting a [Kubernetes ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) onto every Ruler pod.
+
+A typical local configuration might look something like:
+```
+  -Ruler.storage.type=local
+  -Ruler.storage.local.directory=/tmp/loki/rules
+```
+
+With the above configuration, the Ruler would expect the following layout:
+```
+/tmp/loki/rules/<tenant id>/rules1.yaml
+                             /rules2.yaml
+```
+Yaml files are expected to be in the [Prometheus format](#Prometheus_Compatible) but include LogQL expressions as specified in the beginning of this doc.
+
+## Future improvements
+
+There are a few things coming to increase the robustness of this service. In no particular order:
+
+- Recording rules.
+- Backend metric stores adapters for generated alert and recording rule data. The first will likely be Cortex, as Loki is built atop it.
+  - The first will likely be Cortex, as Loki is built atop it.
+- Introduce LogQL v2.
```

Comment:
will this naming be evergreen?

Reply:
I don't know -- this doc will probably need some updates as things progress.